src/shared/redux/modules/member.js

```diff
@@ -37,18 +40,30 @@ const reducer: Reducer<State, Action> =
       case DEACTIVATED_MEMBER:
         return (
           { ...member
-          , activation_status: { value: 'deactivated', initial_value: member.activation_status.initial_value }
+          , activation_status: { value: 'deactivated', initialValue: member.activation_status.initialValue }
           , deletion_date: { value: new Date().toISOString() }
           }
         )
       case REACTIVATED_MEMBER:
         return (
           { ...member
-          , activation_status: { value: 'activated', initial_value: member.activation_status.initial_value }
+          , activation_status: { value: 'activated', initialValue: member.activation_status.initialValue }
           , deletion_date: {}
           , deletion_reason: { value: null }
           }
         )
+      case UPDATED_MEMBERSHIP_TYPE:
```

Comment:
Did you think about having two different cases for the different kind of updates? I could see this being the right way of doing it, but just wondering if you considered both